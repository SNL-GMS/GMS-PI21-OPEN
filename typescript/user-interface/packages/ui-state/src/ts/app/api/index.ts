export * from './data';
export * from './event-manager';
export * from './processing-configuration';
export * from './processing-station';
export * from './signal-enhancement-configuration';
export * from './soh-acei';
export * from './ssam-control';
export * from './station-definition';
export * from './system-event-gateway';
export * from './system-message-definition';
export * from './user-manager';
export * from './util';
export * from './workflow';
