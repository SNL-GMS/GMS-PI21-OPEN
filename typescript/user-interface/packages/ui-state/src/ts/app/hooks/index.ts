export * from './channel-segment-hooks';
export * from './data-acquisition-hooks';
export * from './event-manager-hooks';
export * from './filter-hooks';
export * from './keyboard-shortcuts-hooks';
export * from './operational-time-period-configuration-hooks';
export * from './processing-analyst-configuration-hooks';
export * from './processing-station-hooks';
export * from './query-util-hooks';
export * from './react-redux-hooks';
export * from './signal-detection-hooks';
export * from './station-definition-hooks';
export * from './system-event-gateway-hooks';
export * from './user-manager-hooks';
export * from './waveform-hooks';
