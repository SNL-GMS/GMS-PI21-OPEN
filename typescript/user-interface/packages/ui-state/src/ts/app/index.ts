export * from './api';
export * from './hooks';
export * from './query';
export * from './redux-provider';
export * from './state';
export * from './store';
export * from './subscription';
