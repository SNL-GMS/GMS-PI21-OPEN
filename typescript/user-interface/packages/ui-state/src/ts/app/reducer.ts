import { combineReducers } from '@reduxjs/toolkit';

import {
  eventManagerApiSlice,
  processingConfigurationApiSlice,
  processingStationApiSlice,
  signalEnhancementConfigurationApiSlice,
  sohAceiApiSlice,
  ssamControlApiSlice,
  stationDefinitionSlice,
  systemEventGatewayApiSlice,
  systemMessageDefinitionApiSlice,
  userManagerApiSlice,
  workflowApiSlice
} from './api';
import { dataSlice } from './api/data/data-slice';
import { reducer as appReducer } from './state/reducer';

export const reducer = combineReducers({
  // application api (queries)
  [systemEventGatewayApiSlice.reducerPath]: systemEventGatewayApiSlice.reducer,
  [eventManagerApiSlice.reducerPath]: eventManagerApiSlice.reducer,
  [processingConfigurationApiSlice.reducerPath]: processingConfigurationApiSlice.reducer,
  [processingStationApiSlice.reducerPath]: processingStationApiSlice.reducer,
  [signalEnhancementConfigurationApiSlice.reducerPath]:
    signalEnhancementConfigurationApiSlice.reducer,
  [ssamControlApiSlice.reducerPath]: ssamControlApiSlice.reducer,
  [sohAceiApiSlice.reducerPath]: sohAceiApiSlice.reducer,
  [stationDefinitionSlice.reducerPath]: stationDefinitionSlice.reducer,
  [systemMessageDefinitionApiSlice.reducerPath]: systemMessageDefinitionApiSlice.reducer,
  [userManagerApiSlice.reducerPath]: userManagerApiSlice.reducer,
  [workflowApiSlice.reducerPath]: workflowApiSlice.reducer,
  // data state
  [dataSlice.name]: dataSlice.reducer,
  // application state
  app: appReducer
});
