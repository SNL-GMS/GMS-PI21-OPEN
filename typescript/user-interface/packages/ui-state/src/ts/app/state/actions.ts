import { createAction } from '@reduxjs/toolkit';

/**
 * Action used to reset the application state.
 */
export const reset = createAction('reset');
