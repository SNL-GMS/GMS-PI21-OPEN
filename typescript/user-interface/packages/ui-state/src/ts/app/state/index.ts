export * as AppActions from './actions';
export * from './analyst';
export * from './common';
export * from './data-acquisition';
export * from './events';
export * from './keypress';
export * from './map';
export * as AppOperations from './operations';
export * from './signal-detections';
export * from './station-properties-configuration';
export * from './system-message';
export * from './user-session';
export * from './waveform';
export * from './workflow';
