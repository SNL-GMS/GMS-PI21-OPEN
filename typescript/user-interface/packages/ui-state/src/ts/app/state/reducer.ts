import type { CombinedState } from '@reduxjs/toolkit';
import { combineReducers, createReducer } from '@reduxjs/toolkit';
import type { ReducerWithInitialState } from '@reduxjs/toolkit/dist/createReducer';

import {
  analystInitialState,
  analystSlice,
  commonInitialState,
  commonSlice,
  dataAcquisitionInitialState,
  dataAcquisitionSlice,
  waveformInitialState,
  waveformSlice,
  workflowInitialState,
  workflowSlice
} from '.';
import { reset } from './actions';
import type { AnalystState } from './analyst/types';
import type { CommonState } from './common/types';
import type { DataAcquisitionState } from './data-acquisition/types';
import type { EventsState } from './events';
import { eventsInitialState, eventsSlice } from './events';
import type { MapState } from './map';
import { mapInitialState, mapSlice } from './map';
import type { SignalDetectionsState } from './signal-detections';
import { signalDetectionsInitialState, signalDetectionsSlice } from './signal-detections';
import {
  stationPropertiesConfigurationInitialState,
  stationPropertiesConfigurationSlice
} from './station-properties-configuration';
import type { StationPropertiesConfigurationState } from './station-properties-configuration/types';
import {
  systemMessageInitialState,
  systemMessageSlice
} from './system-message/system-message-slice';
import type { SystemMessageState } from './system-message/types';
import type { UserSessionState } from './user-session/types';
import { userSessionInitialState, userSessionSlice } from './user-session/user-session-slice';
import type { WaveformState } from './waveform/types';
import type { WorkflowState } from './workflow/types';

// combine all reducers
const reducers = combineReducers<{
  analyst: AnalystState;
  stationPropertiesConfiguration: StationPropertiesConfigurationState;
  common: CommonState;
  dataAcquisition: DataAcquisitionState;
  events: EventsState;
  map: MapState;
  signalDetections: SignalDetectionsState;
  systemMessage: SystemMessageState;
  userSession: UserSessionState;
  waveform: WaveformState;
  workflow: WorkflowState;
}>({
  [analystSlice.name]: analystSlice.reducer,
  [stationPropertiesConfigurationSlice.name]: stationPropertiesConfigurationSlice.reducer,
  [commonSlice.name]: commonSlice.reducer,
  [dataAcquisitionSlice.name]: dataAcquisitionSlice.reducer,
  [eventsSlice.name]: eventsSlice.reducer,
  [mapSlice.name]: mapSlice.reducer,
  [signalDetectionsSlice.name]: signalDetectionsSlice.reducer,
  [systemMessageSlice.name]: systemMessageSlice.reducer,
  [userSessionSlice.name]: userSessionSlice.reducer,
  [waveformSlice.name]: waveformSlice.reducer,
  [workflowSlice.name]: workflowSlice.reducer
});

// Infer the `State` type from the reducer itself
type State = ReturnType<typeof reducers>;

/**
 * The app initial state
 */
export const initialState: State = {
  [analystSlice.name]: analystInitialState,
  [stationPropertiesConfigurationSlice.name]: stationPropertiesConfigurationInitialState,
  [commonSlice.name]: commonInitialState,
  [dataAcquisitionSlice.name]: dataAcquisitionInitialState,
  [eventsSlice.name]: eventsInitialState,
  [mapSlice.name]: mapInitialState,
  [signalDetectionsSlice.name]: signalDetectionsInitialState,
  [systemMessageSlice.name]: systemMessageInitialState,
  [userSessionSlice.name]: userSessionInitialState,
  [waveformSlice.name]: waveformInitialState,
  [workflowSlice.name]: workflowInitialState
};

/**
 * The application state reducer
 *
 * Type inference here was causing errors (it was attempting to export types from third party packages).
 * This caused build errors, which is likely an issue with RTK, immer, or typescript itself. Explicitly
 * typing it here seems to make typescript happy.
 */
export const reducer: ReducerWithInitialState<CombinedState<{
  analyst: AnalystState;
  stationPropertiesConfiguration: StationPropertiesConfigurationState;
  common: CommonState;
  dataAcquisition: DataAcquisitionState;
  events: EventsState;
  map: MapState;
  signalDetections: SignalDetectionsState;
  systemMessage: SystemMessageState;
  userSession: UserSessionState;
  waveform: WaveformState;
  workflow: WorkflowState;
}>> = createReducer(initialState, builder => {
  builder
    .addCase(reset, state => {
      state.analyst = initialState.analyst;
      state.stationPropertiesConfiguration = initialState.stationPropertiesConfiguration;
      state.common = initialState.common;
      state.dataAcquisition = initialState.dataAcquisition;
      state.events = initialState.events;
      state.map = initialState.map;
      state.signalDetections = initialState.signalDetections;
      state.systemMessage = initialState.systemMessage;
      state.userSession = initialState.userSession;
      state.waveform = initialState.waveform;
      state.workflow = initialState.workflow;
    })
    .addDefaultCase(reducers);
});
