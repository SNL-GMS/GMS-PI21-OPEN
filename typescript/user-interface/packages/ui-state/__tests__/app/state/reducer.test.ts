import { reset } from '../../../src/ts/app/state/actions';
import { reducer } from '../../../src/ts/app/state/reducer';
import { getStore } from '../../../src/ts/app/store';

describe('state reducer', () => {
  describe('reducer', () => {
    it('should reset the app', () => {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      const store: any = getStore();
      expect(reducer(store.getState(), reset)).toMatchSnapshot();
    });
  });
});
