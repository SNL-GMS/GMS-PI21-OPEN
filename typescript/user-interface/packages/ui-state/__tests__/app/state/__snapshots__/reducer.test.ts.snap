// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`state reducer reducer should reset the app 1`] = `
Object {
  "analyst": Object {
    "alignWaveformsOn": "Time",
    "channelFilters": Object {},
    "defaultSignalDetectionPhase": "P",
    "effectiveNowTime": null,
    "eventListOpenEventTriggered": false,
    "historyActionInProgress": 0,
    "hotkeyCycleOverrides": Object {},
    "location": Object {
      "selectedLocationSolutionId": null,
      "selectedLocationSolutionSetId": null,
      "selectedPreferredLocationSolutionId": null,
      "selectedPreferredLocationSolutionSetId": null,
    },
    "mapOpenEventTriggered": false,
    "measurementMode": Object {
      "entries": Object {},
      "mode": "Default",
    },
    "openEventId": null,
    "openLayoutName": null,
    "phaseToAlignOn": null,
    "sdIdsToShowFk": Array [],
    "selectedEventIds": Array [],
    "selectedFilterIndex": null,
    "selectedFilterList": null,
    "selectedSdIds": Array [],
    "selectedSortType": "Station: A-Z",
  },
  "app": Object {
    "analyst": Object {
      "alignWaveformsOn": "Time",
      "channelFilters": Object {},
      "defaultSignalDetectionPhase": "P",
      "effectiveNowTime": null,
      "eventListOpenEventTriggered": false,
      "historyActionInProgress": 0,
      "hotkeyCycleOverrides": Object {},
      "location": Object {
        "selectedLocationSolutionId": null,
        "selectedLocationSolutionSetId": null,
        "selectedPreferredLocationSolutionId": null,
        "selectedPreferredLocationSolutionSetId": null,
      },
      "mapOpenEventTriggered": false,
      "measurementMode": Object {
        "entries": Object {},
        "mode": "Default",
      },
      "openEventId": null,
      "openLayoutName": null,
      "phaseToAlignOn": null,
      "sdIdsToShowFk": Array [],
      "selectedEventIds": Array [],
      "selectedFilterIndex": null,
      "selectedFilterList": null,
      "selectedSdIds": Array [],
      "selectedSortType": "Station: A-Z",
    },
    "common": Object {
      "commandPaletteIsVisible": false,
      "glLayoutState": Object {},
      "keyPressActionQueue": Object {},
      "keyboardShortcutsVisibility": false,
      "selectedStationIds": Array [],
    },
    "dataAcquisition": Object {
      "data": Object {
        "sohStatus": Object {
          "isStale": true,
          "lastUpdated": 0,
          "loading": true,
          "stationAndStationGroupSoh": Object {
            "isUpdateResponse": false,
            "stationGroups": Array [],
            "stationSoh": Array [],
          },
        },
      },
      "filtersToDisplay": Object {
        "soh-environment-channel-statuses": Object {
          "BAD": true,
          "GOOD": true,
          "MARGINAL": true,
        },
        "soh-environment-monitor-statuses": Object {
          "BAD": true,
          "GOOD": true,
          "MARGINAL": true,
        },
        "soh-lag": Object {
          "BAD": true,
          "GOOD": true,
          "MARGINAL": true,
        },
        "soh-missing": Object {
          "BAD": true,
          "GOOD": true,
          "MARGINAL": true,
        },
        "soh-overview": Object {
          "BAD": true,
          "GOOD": true,
          "MARGINAL": true,
          "NONE": true,
        },
        "soh-timeliness": Object {
          "BAD": true,
          "GOOD": true,
          "MARGINAL": true,
        },
        "station-statistics": Object {
          "BAD": true,
          "GOOD": true,
          "MARGINAL": true,
          "NONE": true,
        },
      },
      "selectedAceiType": null,
      "selectedProcessingStation": null,
      "stationStatisticsGroup": undefined,
      "unmodifiedProcessingStation": null,
    },
    "events": Object {
      "edgeEvents": Object {
        "Edge events after interval": true,
        "Edge events before interval": true,
      },
      "eventsColumns": Object {
        "activeAnalysts": true,
        "confidenceSemiMajorAxis": false,
        "confidenceSemiMinorAxis": false,
        "conflict": true,
        "coverageSemiMajorAxis": false,
        "coverageSemiMinorAxis": false,
        "depthKm": true,
        "latitudeDegrees": true,
        "longitudeDegrees": true,
        "magnitudeMb": true,
        "magnitudeMl": false,
        "magnitudeMs": false,
        "preferred": true,
        "region": true,
        "rejected": true,
        "status": true,
        "time": true,
      },
      "stationsAssociatedWithCurrentOpenEvent": Array [],
    },
    "map": Object {
      "isSyncedWithWaveformZoom": false,
      "layerVisibility": Object {
        "associatedCompleteDetection": true,
        "associatedOpenDetection": true,
        "associatedOtherDetection": true,
        "confidenceEllipse": true,
        "coverageEllipse": true,
        "edgeDetectionAfter": true,
        "edgeDetectionBefore": true,
        "edgeEventsAfterInterval": true,
        "edgeEventsBeforeInterval": true,
        "events": true,
        "nonPreferredLocationSolution": false,
        "preferredLocationSolution": true,
        "signalDetections": true,
        "sites": false,
        "stations": true,
        "unassociatedDetection": true,
      },
    },
    "signalDetections": Object {
      "displayedSignalDetectionConfiguration": Object {
        "signalDetectionAfterInterval": true,
        "signalDetectionAssociatedToCompletedEvent": true,
        "signalDetectionAssociatedToOpenEvent": true,
        "signalDetectionAssociatedToOtherEvent": true,
        "signalDetectionBeforeInterval": true,
        "signalDetectionUnassociated": true,
        "syncWaveform": false,
      },
      "signalDetectionsColumns": Object {
        "amplitude": true,
        "assocStatus": true,
        "azimuth": true,
        "azimuthStandardDeviation": true,
        "channel": true,
        "conflict": true,
        "emergenceAngle": false,
        "longPeriodFirstMotion": false,
        "period": true,
        "phase": true,
        "phaseConfidence": false,
        "rectilinearity": false,
        "rejected": true,
        "sNR": true,
        "shortPeriodFirstMotion": false,
        "slowness": true,
        "slownessStandardDeviation": true,
        "station": true,
        "time": true,
        "timeStandardDeviation": true,
        "unsavedChanges": true,
      },
    },
    "stationPropertiesConfiguration": Object {
      "channelConfigurationColumns": Object {
        "calibrationEffectiveAt": true,
        "calibrationFactor": true,
        "calibrationPeriod": true,
        "calibrationResponseId": true,
        "calibrationStandardDeviation": true,
        "calibrationTimeShift": true,
        "channelBandType": false,
        "channelDataType": false,
        "channelInstrumentType": false,
        "channelOrientationCode": false,
        "channelOrientationType": false,
        "depthKm": true,
        "description": true,
        "eastDisplacementKm": true,
        "effectiveAt": true,
        "effectiveUntil": true,
        "elevationKm": true,
        "fapResponseId": true,
        "latitudeDegrees": true,
        "longitudeDegrees": true,
        "name": true,
        "nominalSampleRateHz": true,
        "northDisplacementKm": true,
        "orientationHorizontalDegrees": true,
        "orientationVerticalDegrees": true,
        "units": true,
        "verticalDisplacementKm": true,
      },
      "channelGroupConfigurationColumns": Object {
        "depthKm": true,
        "description": true,
        "effectiveAt": true,
        "effectiveUntil": true,
        "elevationKm": true,
        "latitudeDegrees": true,
        "longitudeDegrees": true,
        "name": true,
        "type": true,
      },
      "selectedEffectiveAtIndex": 0,
    },
    "systemMessage": Object {
      "isSoundEnabled": true,
      "lastUpdated": null,
      "latestSystemMessages": null,
      "systemMessages": null,
    },
    "userSession": Object {
      "authenticationStatus": Object {
        "authenticated": false,
        "authenticationCheckComplete": false,
        "failedToConnect": false,
        "userName": null,
      },
      "connected": true,
    },
    "waveform": Object {
      "baseStationTime": 0,
      "loadingState": Object {
        "completed": 0,
        "description": "Loading waveforms",
        "isLoading": false,
        "percent": 0,
        "total": 0,
      },
      "maximumOffset": 0,
      "minimumOffset": 0,
      "shouldShowPredictedPhases": true,
      "shouldShowTimeUncertainty": false,
      "stationsVisibility": Object {},
      "viewableInterval": null,
      "zoomInterval": null,
    },
    "workflow": Object {
      "analysisMode": null,
      "openActivityNames": Array [],
      "openIntervalName": null,
      "stationGroup": null,
      "timeRange": Object {
        "endTimeSecs": null,
        "startTimeSecs": null,
      },
    },
  },
  "common": Object {
    "commandPaletteIsVisible": false,
    "glLayoutState": Object {},
    "keyPressActionQueue": Object {},
    "keyboardShortcutsVisibility": false,
    "selectedStationIds": Array [],
  },
  "data": Object {
    "events": Object {},
    "queries": Object {
      "getChannelSegmentsByChannel": Object {},
      "getEventsWithDetectionsAndSegmentsByTime": Object {},
      "getSignalDetectionWithSegmentsByStationAndTime": Object {},
    },
    "signalDetections": Object {},
    "uiChannelSegments": Object {},
  },
  "dataAcquisition": Object {
    "data": Object {
      "sohStatus": Object {
        "isStale": true,
        "lastUpdated": 0,
        "loading": true,
        "stationAndStationGroupSoh": Object {
          "isUpdateResponse": false,
          "stationGroups": Array [],
          "stationSoh": Array [],
        },
      },
    },
    "filtersToDisplay": Object {
      "soh-environment-channel-statuses": Object {
        "BAD": true,
        "GOOD": true,
        "MARGINAL": true,
      },
      "soh-environment-monitor-statuses": Object {
        "BAD": true,
        "GOOD": true,
        "MARGINAL": true,
      },
      "soh-lag": Object {
        "BAD": true,
        "GOOD": true,
        "MARGINAL": true,
      },
      "soh-missing": Object {
        "BAD": true,
        "GOOD": true,
        "MARGINAL": true,
      },
      "soh-overview": Object {
        "BAD": true,
        "GOOD": true,
        "MARGINAL": true,
        "NONE": true,
      },
      "soh-timeliness": Object {
        "BAD": true,
        "GOOD": true,
        "MARGINAL": true,
      },
      "station-statistics": Object {
        "BAD": true,
        "GOOD": true,
        "MARGINAL": true,
        "NONE": true,
      },
    },
    "selectedAceiType": null,
    "selectedProcessingStation": null,
    "stationStatisticsGroup": undefined,
    "unmodifiedProcessingStation": null,
  },
  "eventManagerApi": Object {
    "config": Object {
      "focused": true,
      "keepUnusedDataFor": 60,
      "middlewareRegistered": false,
      "online": true,
      "reducerPath": "eventManagerApi",
      "refetchOnFocus": false,
      "refetchOnMountOrArgChange": false,
      "refetchOnReconnect": false,
    },
    "mutations": Object {},
    "provided": Object {},
    "queries": Object {},
    "subscriptions": Object {},
  },
  "events": Object {
    "edgeEvents": Object {
      "Edge events after interval": true,
      "Edge events before interval": true,
    },
    "eventsColumns": Object {
      "activeAnalysts": true,
      "confidenceSemiMajorAxis": false,
      "confidenceSemiMinorAxis": false,
      "conflict": true,
      "coverageSemiMajorAxis": false,
      "coverageSemiMinorAxis": false,
      "depthKm": true,
      "latitudeDegrees": true,
      "longitudeDegrees": true,
      "magnitudeMb": true,
      "magnitudeMl": false,
      "magnitudeMs": false,
      "preferred": true,
      "region": true,
      "rejected": true,
      "status": true,
      "time": true,
    },
    "stationsAssociatedWithCurrentOpenEvent": Array [],
  },
  "map": Object {
    "isSyncedWithWaveformZoom": false,
    "layerVisibility": Object {
      "associatedCompleteDetection": true,
      "associatedOpenDetection": true,
      "associatedOtherDetection": true,
      "confidenceEllipse": true,
      "coverageEllipse": true,
      "edgeDetectionAfter": true,
      "edgeDetectionBefore": true,
      "edgeEventsAfterInterval": true,
      "edgeEventsBeforeInterval": true,
      "events": true,
      "nonPreferredLocationSolution": false,
      "preferredLocationSolution": true,
      "signalDetections": true,
      "sites": false,
      "stations": true,
      "unassociatedDetection": true,
    },
  },
  "processingConfigurationApi": Object {
    "config": Object {
      "focused": true,
      "keepUnusedDataFor": 60,
      "middlewareRegistered": false,
      "online": true,
      "reducerPath": "processingConfigurationApi",
      "refetchOnFocus": false,
      "refetchOnMountOrArgChange": false,
      "refetchOnReconnect": false,
    },
    "mutations": Object {},
    "provided": Object {},
    "queries": Object {},
    "subscriptions": Object {},
  },
  "processingStationApi": Object {
    "config": Object {
      "focused": true,
      "keepUnusedDataFor": 60,
      "middlewareRegistered": false,
      "online": true,
      "reducerPath": "processingStationApi",
      "refetchOnFocus": false,
      "refetchOnMountOrArgChange": false,
      "refetchOnReconnect": false,
    },
    "mutations": Object {},
    "provided": Object {},
    "queries": Object {},
    "subscriptions": Object {},
  },
  "signalDetections": Object {
    "displayedSignalDetectionConfiguration": Object {
      "signalDetectionAfterInterval": true,
      "signalDetectionAssociatedToCompletedEvent": true,
      "signalDetectionAssociatedToOpenEvent": true,
      "signalDetectionAssociatedToOtherEvent": true,
      "signalDetectionBeforeInterval": true,
      "signalDetectionUnassociated": true,
      "syncWaveform": false,
    },
    "signalDetectionsColumns": Object {
      "amplitude": true,
      "assocStatus": true,
      "azimuth": true,
      "azimuthStandardDeviation": true,
      "channel": true,
      "conflict": true,
      "emergenceAngle": false,
      "longPeriodFirstMotion": false,
      "period": true,
      "phase": true,
      "phaseConfidence": false,
      "rectilinearity": false,
      "rejected": true,
      "sNR": true,
      "shortPeriodFirstMotion": false,
      "slowness": true,
      "slownessStandardDeviation": true,
      "station": true,
      "time": true,
      "timeStandardDeviation": true,
      "unsavedChanges": true,
    },
  },
  "signalEnhancementConfigurationApi": Object {
    "config": Object {
      "focused": true,
      "keepUnusedDataFor": 60,
      "middlewareRegistered": false,
      "online": true,
      "reducerPath": "signalEnhancementConfigurationApi",
      "refetchOnFocus": false,
      "refetchOnMountOrArgChange": false,
      "refetchOnReconnect": false,
    },
    "mutations": Object {},
    "provided": Object {},
    "queries": Object {},
    "subscriptions": Object {},
  },
  "sohAceiApi": Object {
    "config": Object {
      "focused": true,
      "keepUnusedDataFor": 60,
      "middlewareRegistered": false,
      "online": true,
      "reducerPath": "sohAceiApi",
      "refetchOnFocus": false,
      "refetchOnMountOrArgChange": false,
      "refetchOnReconnect": false,
    },
    "mutations": Object {},
    "provided": Object {},
    "queries": Object {},
    "subscriptions": Object {},
  },
  "ssamControlApi": Object {
    "config": Object {
      "focused": true,
      "keepUnusedDataFor": 60,
      "middlewareRegistered": false,
      "online": true,
      "reducerPath": "ssamControlApi",
      "refetchOnFocus": false,
      "refetchOnMountOrArgChange": false,
      "refetchOnReconnect": false,
    },
    "mutations": Object {},
    "provided": Object {},
    "queries": Object {},
    "subscriptions": Object {},
  },
  "stationDefinitionApi": Object {
    "config": Object {
      "focused": true,
      "keepUnusedDataFor": 60,
      "middlewareRegistered": false,
      "online": true,
      "reducerPath": "stationDefinitionApi",
      "refetchOnFocus": false,
      "refetchOnMountOrArgChange": false,
      "refetchOnReconnect": false,
    },
    "mutations": Object {},
    "provided": Object {},
    "queries": Object {},
    "subscriptions": Object {},
  },
  "stationPropertiesConfiguration": Object {
    "channelConfigurationColumns": Object {
      "calibrationEffectiveAt": true,
      "calibrationFactor": true,
      "calibrationPeriod": true,
      "calibrationResponseId": true,
      "calibrationStandardDeviation": true,
      "calibrationTimeShift": true,
      "channelBandType": false,
      "channelDataType": false,
      "channelInstrumentType": false,
      "channelOrientationCode": false,
      "channelOrientationType": false,
      "depthKm": true,
      "description": true,
      "eastDisplacementKm": true,
      "effectiveAt": true,
      "effectiveUntil": true,
      "elevationKm": true,
      "fapResponseId": true,
      "latitudeDegrees": true,
      "longitudeDegrees": true,
      "name": true,
      "nominalSampleRateHz": true,
      "northDisplacementKm": true,
      "orientationHorizontalDegrees": true,
      "orientationVerticalDegrees": true,
      "units": true,
      "verticalDisplacementKm": true,
    },
    "channelGroupConfigurationColumns": Object {
      "depthKm": true,
      "description": true,
      "effectiveAt": true,
      "effectiveUntil": true,
      "elevationKm": true,
      "latitudeDegrees": true,
      "longitudeDegrees": true,
      "name": true,
      "type": true,
    },
    "selectedEffectiveAtIndex": 0,
  },
  "systemEventGatewayApi": Object {
    "config": Object {
      "focused": true,
      "keepUnusedDataFor": 60,
      "middlewareRegistered": false,
      "online": true,
      "reducerPath": "systemEventGatewayApi",
      "refetchOnFocus": false,
      "refetchOnMountOrArgChange": false,
      "refetchOnReconnect": false,
    },
    "mutations": Object {},
    "provided": Object {},
    "queries": Object {},
    "subscriptions": Object {},
  },
  "systemMessage": Object {
    "isSoundEnabled": true,
    "lastUpdated": null,
    "latestSystemMessages": null,
    "systemMessages": null,
  },
  "systemMessageDefinitionApi": Object {
    "config": Object {
      "focused": true,
      "keepUnusedDataFor": 60,
      "middlewareRegistered": false,
      "online": true,
      "reducerPath": "systemMessageDefinitionApi",
      "refetchOnFocus": false,
      "refetchOnMountOrArgChange": false,
      "refetchOnReconnect": false,
    },
    "mutations": Object {},
    "provided": Object {},
    "queries": Object {},
    "subscriptions": Object {},
  },
  "userManagerApi": Object {
    "config": Object {
      "focused": true,
      "keepUnusedDataFor": 60,
      "middlewareRegistered": false,
      "online": true,
      "reducerPath": "userManagerApi",
      "refetchOnFocus": false,
      "refetchOnMountOrArgChange": false,
      "refetchOnReconnect": false,
    },
    "mutations": Object {},
    "provided": Object {},
    "queries": Object {},
    "subscriptions": Object {},
  },
  "userSession": Object {
    "authenticationStatus": Object {
      "authenticated": false,
      "authenticationCheckComplete": false,
      "failedToConnect": false,
      "userName": null,
    },
    "connected": true,
  },
  "waveform": Object {
    "baseStationTime": 0,
    "loadingState": Object {
      "completed": 0,
      "description": "Loading waveforms",
      "isLoading": false,
      "percent": 0,
      "total": 0,
    },
    "maximumOffset": 0,
    "minimumOffset": 0,
    "shouldShowPredictedPhases": true,
    "shouldShowTimeUncertainty": false,
    "stationsVisibility": Object {},
    "viewableInterval": null,
    "zoomInterval": null,
  },
  "workflow": Object {
    "analysisMode": null,
    "openActivityNames": Array [],
    "openIntervalName": null,
    "stationGroup": null,
    "timeRange": Object {
      "endTimeSecs": null,
      "startTimeSecs": null,
    },
  },
  "workflowApi": Object {
    "config": Object {
      "focused": true,
      "keepUnusedDataFor": 60,
      "middlewareRegistered": false,
      "online": true,
      "reducerPath": "workflowApi",
      "refetchOnFocus": false,
      "refetchOnMountOrArgChange": false,
      "refetchOnReconnect": false,
    },
    "mutations": Object {},
    "provided": Object {},
    "queries": Object {},
    "subscriptions": Object {},
  },
}
`;
